src/Microsoft.Diagnostics.NETCore.Client/ReversedServer/ReversedDiagnosticsServer.cs

```diff
@@ -69,12 +70,38 @@ public ReversedDiagnosticsServer(string address)
         /// Otherwise if kind is TcpIp as a supported protocol for ReversedDiagnosticServer. When Kind is Tcp, address will
         /// be analyzed and if on format host:port, ReversedDiagnosticServer will try to bind
         /// a TcpIp listener to host and port, otherwise it will use a Unix domain socket or a Windows named pipe.
-        ///
         /// </param>
         public ReversedDiagnosticsServer(string address, Kind kind)
         {
             _address = address;
             _kind = kind;
+            ParseAdvertiseTimeout = TimeSpan.FromMilliseconds(750);
+        }
+
+        /// <summary>
+        /// Constructs the <see cref="ReversedDiagnosticsServer"/> instance with an endpoint bound
+        /// to the location specified by <paramref name="address"/>.
+        /// </summary>
+        /// <param name="address">
+        /// The server endpoint.
+        /// On Windows, this can be a full pipe path or the name without the "\\.\pipe\" prefix.
+        /// On all other systems, this must be the full file path of the socket.
+        /// </param>
+        /// <param name="kind">
+        /// If kind is WebSocket, start a Kestrel web server.
+        /// Otherwise if kind is TcpIp as a supported protocol for ReversedDiagnosticServer. When Kind is Tcp, address will
+        /// be analyzed and if on format host:port, ReversedDiagnosticServer will try to bind
+        /// a TcpIp listener to host and port, otherwise it will use a Unix domain socket or a Windows named pipe.
+        /// </param>
+        /// <param name="timeout">
+        /// The amount of time to allow parsing of the advertise data before cancelling. This allows the server to
+        /// remain responsive in case the advertise data is incomplete and the stream is not closed.
+        /// </param>
+        public ReversedDiagnosticsServer(string address, Kind kind, TimeSpan timeout)
```

Comment:
Change web socket implementation to instance ReversedDiagnosticsServer with this and its default timeout.